third_party/USB-UIRT/Transmit/uuirtdrv.h

```diff
@@ -0,0 +1,137 @@
+//public:
+//	CUuirtdrv(void);
+//	// TODO: add your methods here.
+//};
+
+typedef struct
+{
+	unsigned int fwVersion;
+	unsigned int protVersion;
+	unsigned char fwDateDay;
+	unsigned char fwDateMonth;
+	unsigned char fwDateYear;
+} UUINFO, *PUUINFO;
+
+typedef struct UUGPIO
+{
+	BYTE	irCode[6];
+	BYTE	action;
+	BYTE	duration;
+} *PUUGPIO;
+
+typedef void *HUUHANDLE;
+
+typedef void (WINAPI *PUUCALLBACKPROC) (char *IREventStr, void *userData);
+typedef void (WINAPI *PLEARNCALLBACKPROC) (unsigned int progress, unsigned int sigQuality, unsigned long carrierFreq, void *userData);
+
+//typedef void (__stdcall *PUUCALLBACKPROC)(BSTR pbstr);
+
+
+extern UUIRTDRV_API int nUuirtdrv;
+
+UUIRTDRV_API HUUHANDLE PASCAL UUIRTOpen(void);
+
+UUIRTDRV_API HUUHANDLE PASCAL UUIRTOpenEx(char *devName, DWORD dwFlagsAndAttributes, void *reserved0, void *reserved1);
+
+#define UUIRTDRV_OPENEX_ATTRIBUTE_EXCLUSIVE	0x0001
+
+UUIRTDRV_API BOOL PASCAL UUIRTClose(HUUHANDLE hHandle);
+
+UUIRTDRV_API BOOL PASCAL UUIRTGetDrvInfo(unsigned int *puAPIVersion);
+//UUIRTDRV_API BOOL PASCAL UUIRTGetDrvInfo2(LPDWORD puDrvVersion);
+
+UUIRTDRV_API BOOL PASCAL UUIRTGetDrvVersion(unsigned int *puDLLVersion);
+
+UUIRTDRV_API BOOL PASCAL UUIRTGetUUIRTInfo(HUUHANDLE hHandle, PUUINFO puuInfo);
+
+UUIRTDRV_API BOOL PASCAL UUIRTGetUUIRTConfig(HUUHANDLE hHandle, PUINT32 puConfig);
+
+UUIRTDRV_API BOOL PASCAL UUIRTSetUUIRTConfig(HUUHANDLE hHandle, UINT32 uConfig);
+
+UUIRTDRV_API BOOL PASCAL UUIRTTransmitIR(HUUHANDLE hHandle, char *IRCode, int codeFormat, int repeatCount, int inactivityWaitTime, HANDLE hEvent, void *reserved0, void *reserved1);
+
+UUIRTDRV_API BOOL PASCAL UUIRTLearnIR(HUUHANDLE hHandle, int codeFormat, char *IRCode, PLEARNCALLBACKPROC progressProc, void *userData, BOOL *pAbort, unsigned int param1, HANDLE hEvent, void *reserved0);
+
+UUIRTDRV_API BOOL PASCAL UUIRTSetReceiveCallback(HUUHANDLE hHandle, PUUCALLBACKPROC receiveProc, void *userData);
+
+UUIRTDRV_API BOOL PASCAL UUIRTGetUUIRTGPIOCfg(HUUHANDLE hHandle, int *pNumSlots, UINT32 *pdwPortPins, PUUGPIO pGPIOStruct);
+
+UUIRTDRV_API BOOL PASCAL UUIRTSetUUIRTGPIOCfg(HUUHANDLE hHandle, int index, PUUGPIO pGpioSt);
+
+#if 0
+UUIRTDRV_API BOOL PASCAL UUIRTEERead(HUUHANDLE hHandle, FT_PROGRAM_DATA *ftData);
+
+UUIRTDRV_API BOOL PASCAL UUIRTEEProgram(HUUHANDLE hHandle, FT_PROGRAM_DATA *ftData);
+#endif
+
+#define UUIRTDRV_ERR_NO_DEVICE 0x20000001
+#define UUIRTDRV_ERR_NO_RESP   0x20000002
+#define UUIRTDRV_ERR_NO_DLL    0x20000003
+#define UUIRTDRV_ERR_VERSION   0x20000004
+#define UUIRTDRV_ERR_IN_USE    0x20000005
+
+
+
+typedef HUUHANDLE	(WINAPI *pfn_UUIRTOpen)		(void);
+typedef HUUHANDLE	(WINAPI *pfn_UUIRTOpenEx)		(char *devName, DWORD dwFlagsAndAttributes, void *reserved0, void *reserved1);
+typedef BOOL		(WINAPI *pfn_UUIRTClose)	(HUUHANDLE hHandle);
+typedef BOOL		(WINAPI *pfn_UUIRTGetDrvInfo)(unsigned int *puDrvVersion);
+typedef	BOOL		(WINAPI *pfn_UUIRTGetDrvVersion)(unsigned int *puDLLVersion);
```

Comment:
Various alignment/indentation issues to fix below if you want to.

Reply:
This is the vendor-provided include file for the driver DLL.  More TAB victims.  ;)